src/libraries/Microsoft.Extensions.Configuration.Binder/gen/Emitter/CoreBindingHelpers.cs

```diff
@@ -1110,24 +1109,24 @@ private bool EmitObjectInit(ComplexTypeSpec type, string memberAccessExpr, Initi
                 string? castExpr = null;
                 string initExpr;
 
-                string effectiveDisplayString = type.DisplayString;
+                string typeFQN = type.TypeRef.FullyQualifiedName;
                 if (collectionType is not null)
                 {
                     if (collectionType is ArraySpec)
                     {
-                        initExpr = $"new {s_arrayBracketsRegex.Replace(effectiveDisplayString, "[0]", 1)}";
+                        initExpr = $"new {s_arrayBracketsRegex.Replace(typeFQN, "[0]", 1)}";
                     }
                     else
                     {
                         CollectionWithCtorInitSpec collectionWithCtorInitType = (CollectionWithCtorInitSpec)collectionType;
 
                         if (collectionWithCtorInitType.InstantiationConcreteType is not CollectionInstantiationConcreteType.Self)
                         {
-                            castExpr = $"({collectionWithCtorInitType.DisplayString})";
+                            castExpr = $"({collectionWithCtorInitType.TypeRef.FullyQualifiedName})";
                         }
 
-                        effectiveDisplayString = _typeIndex.GetInstantiationTypeDisplayString(collectionWithCtorInitType);
-                        initExpr = $"{castExpr}new {effectiveDisplayString}()";
+                        typeFQN = TypeIndex.GetInstantiationTypeDisplayString(collectionWithCtorInitType);
```

Comment:
>TypeIndex [](http://example.com/codeflow?start=34&length=9)

why didn't use `_typeIndex` here?

Reply:
Because the implementation changed to use the FQN, the method doesn't require any indirection and was converted to `static`.